code/modules/admin/admin.dm

```diff
@@ -31,6 +31,32 @@ proc/admin_notice(var/message, var/rights)
 
 ///////////////////////////////////////////////////////////////////////////////////////////////Panels
 
+/datum/admins/proc/view_log_panel(mob/M)
+	if(!M)
+		to_chat(usr, "That mob doesn't seem to exist! Something went wrong.")
+		return 
+
+	if (!istype(src, /datum/admins))
+		src = usr.client.holder
+	if (!istype(src, /datum/admins))
+		usr << "Error: you are not an admin!"
+		return
+
+	var/body = "<html><head><title>Log Panel of [M.real_name]</title></head>"
+	body += "<body><center>Logs of <b>[M]</b><br>"
+	body += "<a href='?src=\ref[src];viewlogs=\ref[M]'>REFRESH</a></center><br>"
+
+
+	var/i = length(M.attack_log)
+	while(i > 0)
+		body += M.attack_log[i] + "<br>"
+		i--
+
+	usr << browse(body, "window=\ref[M]logs;size=500x500")
```

Comment:
The ref is there so you can have multiple windows open at once.

Reply:
don't we have user_browse or something like that?
Pure output without wrapping is not so good